File: 377-combination-sum-iv/377-combination-sum-iv.cpp
class Solution
{
    public:
    int combinationSum4(vector<int> &nums, int target)
    {
        int n=nums.size();
        sort(nums.begin(),nums.end());
        unsigned int dp[target+1];
        memset(dp,0,sizeof(dp));
        dp[0]=1;
        for(int i=1;i<=target;i++)
        {
            for(auto j:nums)
            {
                if(i-j>=0)
                {
                    dp[i]+=dp[i-j];
                }
                else
                {
                    break;
                }
            }
        }
        return dp[target];
    }
};

File: 377-combination-sum-iv/377-combination-sum-iv.py
class Solution:
    def combinationSum4(self, nums: List[int], target: int) -> int:
        n=len(nums)
        dp=[0]*(target+1)
        dp[0]=1
        nums.sort()
        for i in range(1,target+1):
            for j in nums:
                if i-j>=0:
                    dp[i]+=dp[i-j]
                else:
                    break
        return dp[target]            
        